lib/fencing/st_output.c

```diff
@@ -76,35 +76,60 @@ stonith__history_description(stonith_history_t *history, bool full_history,
 {
     GString *str = g_string_sized_new(256); // Generous starting size
     char *retval = NULL;
+    char *completed_time = NULL;
 
-    g_string_printf(str, "%s of %s %s",
-                    stonith_action_str(history->action), history->target,
-                    state_str(history));
+    if ((history->state == st_failed) || (history->state == st_done)) {
+        completed_time = time_t_string(history->completed);
+    }
+
+    g_string_printf(str, "%s of %s",
+                    stonith_action_str(history->action), history->target);
+
+    if (!pcmk_is_set(show_opts, pcmk_show_failed_detail)) {
+        // More human-friendly
+        if (((history->state == st_failed) || (history->state == st_done))
+            && (history->delegate != NULL)) {
+            g_string_append_printf(str, " by %s", history->delegate);
+        }
+        g_string_append_printf(str, " for %s@%s",
+                               history->client, history->origin);
+        if (!full_history) {
+            g_string_append(str, " last"); // For example, "last failed at ..."
+        }
+    }
+
+    g_string_append_printf(str, " %s", state_str(history));
 
     // For failed actions, add exit reason if available
     if ((history->state == st_failed) && (history->exit_reason != NULL)) {
         g_string_append_printf(str, " (%s)", history->exit_reason);
     }
 
-    g_string_append(str, ": ");
+    if (pcmk_is_set(show_opts, pcmk_show_failed_detail)) {
+        // More technical
+        g_string_append(str, ": ");
 
-    // For completed actions, add delegate if available
-    if (((history->state == st_failed) || (history->state == st_done))
-        && (history->delegate != NULL)) {
-        g_string_append_printf(str, "delegate=%s, ", history->delegate);
-    }
+        // For completed actions, add delegate if available
+        if (((history->state == st_failed) || (history->state == st_done))
+            && (history->delegate != NULL)) {
 
-    // Add information about originator
-    g_string_append_printf(str, "client=%s, origin=%s",
-                           history->client, history->origin);
+            g_string_append_printf(str, "delegate=%s, ", history->delegate);
+        }
 
-    // For completed actions, add completion time
-    if ((history->state == st_failed) || (history->state == st_done)) {
-        char *completed_time = time_t_string(history->completed);
+        // Add information about originator
+        g_string_append_printf(str, "client=%s, origin=%s",
+                               history->client, history->origin);
 
-        g_string_append(str, (full_history? ", completed" : ", last-failed"));
-        g_string_append_printf(str, "='%s'", completed_time);
-        free(completed_time);
+        // For completed actions, add completion time
+        if (completed_time != NULL) {
+            g_string_append(str,
+                            (full_history? ", completed" : ", last-failed"));
```

Comment:
wouldn't we still get in here in the success-case so that we should have something like (instead of 2 ternary operators nested):

if (full_history) {
    g_string_append(str, ", completed");
} else {
    g_string_append(str, (history->state == failed)?", last-failed":", last-successful");
}

Reply:
Ah of course, I see now. Updated commits